include/slate/BaseMatrix.hh

```diff
@@ -1785,6 +1792,43 @@ void BaseMatrix<scalar_t>::tileSend(
     }
 }
 
+//------------------------------------------------------------------------------
+/// Immediately send tile {i, j} of op(A) to the given MPI rank.
+/// Destination rank must call tileRecv().
+///
+/// @tparam target
+///     Destination to target; either Host (default) or Device.
+///
+/// @param[in] i
+///     Tile's block row index. 0 <= i < mt.
+///
+/// @param[in] j
+///     Tile's block column index. 0 <= j < nt.
+///
+/// @param[in] dst_rank
+///     Destination MPI rank. If dst_rank == mpiRank, this is a no-op.
+///
+/// @param[in] tag
+///     MPI tag, default 0.
+///
+/// @param[out] request
+///     Pointer to an MPI_Request struct
+///
+template <typename scalar_t>
+template <Target target>
+void BaseMatrix<scalar_t>::tileIsend(
+    int64_t i, int64_t j, int dst_rank, int tag, MPI_Request* request)
+{
+    if (dst_rank != mpiRank()) {
+        //todo: need to qcquire read access lock to TileNode(i, j)
+        tileGetForReading(i, j, LayoutConvert::None);
```

Comment:
Stupid question: why is it on the host and not from a device as well?

Reply:
Because that's how `tileSend` does it.  This is basically the exact same code, except we have a `request` object.

Reply:
So, maybe for later, but should we have `tileSend` calling `tileIsend` then?

Reply:
Yea, that'd be a good call.  I'll put it on my list of things to add to #122.